### PR TITLE
[Wallet] Masternodes start performance boost

### PR DESCRIPTION
**Pull request by furszy**
Problem:

Essentially, every time that the controller started a masternode, it was looping over the entire set of transactions and outputs to get the available ones, then looping again for the specific collateral utxo of the masternode that needs to be started.
so.. if the wallet had a big number of transactions, the start all had to process an excessive amount of utxo for each masternode being started Making the start all command slow as hell.

Solution:

`GetMasternodeVinAndKeys` method rewritten to directly find the collateral utxo in the wallet's map, perform the availability validations and return the needed keys. Zero loops.

### CODE REVIEW

**Comment by furszy** [comment]
Good feedback 👍 . We were actually not checking for the mind `MASTERNODE_MIN_CONFIRMATIONS` in the wallet before.
Changes applied.

File: src/wallet/wallet.cpp [hunk 1]
```diff
@@ -2002,6 +1962,80 @@ void CWallet::GetAvailableP2CSCoins(std::vector<COutput>& vCoins) const {
...
+    }
+
+    // Find specific vin
+    uint256 txHash = uint256S(strTxHash);
+    std::map<uint256, CWalletTx>::const_iterator mi = mapWallet.find(txHash);
+    if (mi == mapWallet.end()) {
+        return error("%s: tx hash %s not found in the wallet", __func__, strTxHash);
+    }
+    const CWalletTx& wtx = mi->second;
+    CTxOut txOut = wtx.vout[nOutputIndex];
```
Comment (random-zebra) [inline src/wallet/wallet.cpp:2008]: might be worth to check the output index before the assignment of `txOut`:
```cpp
if (nOutputIndex < 0 || nOutputIndex >= wtx.vout.size())
     return error("...");
```

**Comment by furszy** [comment]
Great finding!, let's clean all of these abominations in a subsequent PR/s.

Going to add the output index range check and this should be ready to go.

File: src/wallet/wallet.cpp (hunk 1 above)
Comment (furszy) [reply src/wallet/wallet.cpp:2014]: good, added.

**Pull request merged**